src/fetchBaseQuery.ts

```diff
@@ -2,18 +2,35 @@ import { joinUrls } from './utils';
 import { isPlainObject } from '@reduxjs/toolkit';
 import { BaseQueryFn } from './apiTypes';
 
+export type ResponseHandler = 'json' | 'text' | ((response: Response) => Promise<any>);
+
 export interface FetchArgs extends RequestInit {
   url: string;
   params?: Record<string, any>;
   body?: any;
-  responseHandler?: 'json' | 'text' | ((response: Response) => Promise<any>);
+  responseHandler?: ResponseHandler;
   validateStatus?: (response: Response, body: any) => boolean;
 }
 
 const defaultValidateStatus = (response: Response) => response.status >= 200 && response.status <= 299;
 
 const isJsonContentType = (headers: Headers) => headers.get('content-type')?.trim()?.startsWith('application/json');
 
+const handleResponse = async (response: Response, responseHandler: ResponseHandler) => {
+  if (typeof responseHandler === 'function') {
+    return responseHandler(response);
+  }
+
+  if (responseHandler === 'text') {
+    return response.text();
+  }
+
+  if (responseHandler === 'json') {
+    const text = await response.text();
+    return text.length ? JSON.parse(text) : null;
```

Comment:
Hmm.
Not sure what this should be.
I'm tending more towards `undefined`.
@msutkowski @markerikson can you take a look?

Reply:
Yeah, I can definitely see the case for this being `undefined` instead. Happy to update once the others weigh in.

Reply:
undefined is probably better... the body was technically undefined in this case, ya?